File: services/crm-service/src/main/java/com/crm/crm_backend/model/entity/Lead.java
package com.crm.crm_backend.model.entity;

// model/entity/Lead.java

import com.crm.crm_backend.common.BaseEntity;
import com.crm.crm_backend.model.enums.LeadRating;
import com.crm.crm_backend.model.enums.LeadSource;
import com.crm.crm_backend.model.enums.LeadStatus;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "crm_leads")
public class Lead extends BaseEntity {

    private String firstName;
    private String lastName;
    private String middleName;

    @Column(unique = true)
    private String email;

    private String phone;

    private String company;
    private String companySize;
    private String industry;
    private String jobTitle;
    private String department;

    @Enumerated(EnumType.STRING)
    private LeadSource source;

    private String sourceDetails;

    @Enumerated(EnumType.STRING)
    private LeadStatus status = LeadStatus.NEW;

    @Enumerated(EnumType.STRING)
    private LeadRating rating = LeadRating.WARM;

    private Integer leadScore = 0;

    private BigDecimal budget;
    private String authority;
    private String need;
    private String timeline;
    private Integer qualificationScore;

    private String assignedTo;
    private String assignedTeam;
    private Long campaignId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "territory_id")
    private Territory territory;

    private String preferredContactMethod;
    private String bestTimeToContact;

    @Lob
    private String notes;
    @Lob
    private String internalNotes;

    @ManyToOne
    @JoinColumn(name = "converted_customer_id")
    private Customer convertedCustomer;

    private Long convertedOpportunityId;
    private LocalDateTime convertedAt;
    private String conversionReason;

    private String dataClassification = "CONFIDENTIAL";
    private String securityClearance = "LEVEL_1";
}

File: frontend/src/features/crm/types/lead.ts
export type LeadStatus =
  | "NEW"
  | "CONTACTED"
  | "QUALIFIED"
  | "UNQUALIFIED"
  | "LOST"
  | "CONVERTED";

export type LeadSource =
  | "WEBSITE"
  | "REFERRAL"
  | "COLD_CALL"
  | "SOCIAL_MEDIA"
  | "EVENT"
  | "PARTNER";

export type Lead = {
  id: number;
  firstName: string;
  lastName: string;
  middleName: string | null;
  email: string | null;
  phone: string | null;
  company: string | null;
  industry: string | null;
  jobTitle: string | null;
  source: LeadSource | null;
  sourceDetails: string | null;
  status: LeadStatus;
  leadScore: number | null;
  assignedTo: string | null;
  notes: string | null;
  territoryId: number | null;
  convertedCustomerId: number | null;
  convertedOpportunityId: number | null;
  convertedAt: string | null;
  createdAt: string | null;
  updatedAt: string | null;
};

export type LeadCreateInput = {
  firstName: string;
  lastName: string;
  email?: string;
  phone?: string;
  company?: string;
  industry?: string;
  jobTitle?: string;
  source?: LeadSource;
  sourceDetails?: string;
  assignedTo?: string;
  notes?: string;
};

export type LeadConvertInput = {
  conversionReason?: string;
  createOpportunity?: boolean;
  opportunityValue?: number;
  opportunityTitle?: string;
};

export function leadDisplayId(id: number): string {
  return `LED-${String(id).padStart(6, "0")}`;
}

export function leadFullName(lead: Lead): string {
  return [lead.firstName, lead.lastName].filter(Boolean).join(" ");
}
